src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/types.js

```diff
@@ -14,6 +14,7 @@ export type ToolbarProps = {
     storeKey?: string,
     onNavigationButtonClick?: () => void,
     navigationOpen?: boolean,
+    navigationPinned?: boolean,
```

Comment:
I think from a `Toolbar` point of view, it doesn't matter if the navigation is pinned or not, just if the navigation should be shown or not. And until now we have defined that by checking if something has been passed to the `onNavigationButtonClick` prop. So instead of introducing this boolean, I would not pass a callback to the `onNavigationButtonClick` prop if the navigation is pinned.